rust/libnewsboat-ffi/src/configcontainer.rs

```diff
@@ -0,0 +1,187 @@
+        type ConfigContainer;
+
+        fn create() -> Box<ConfigContainer>;
+
+        fn handle_action(
+            cc: &ConfigContainer,
+            action: &str,
+            params: Vec<String>,
+        ) -> CfgHandlerResult;
+
+        fn get_configvalue(cc: &ConfigContainer, key: &str) -> String;
+        fn get_configvalue_as_int(cc: &ConfigContainer, key: &str) -> i32;
+        fn get_configvalue_as_bool(cc: &ConfigContainer, key: &str) -> bool;
+        fn get_configvalue_as_filepath(
+            cc: &ConfigContainer,
+            key: &str,
+            path: Pin<&mut PathBuf>,
+        ) -> bool;
+
+        fn set_configvalue(
+            cc: &ConfigContainer,
+            key: &str,
+            value: &str,
+            error_msg: &mut String,
+        ) -> bool;
+        fn reset_to_default(cc: &ConfigContainer, key: &str);
+        fn toggle(cc: &ConfigContainer, key: &str);
+        fn dump_config(cc: &ConfigContainer) -> Vec<String>;
+        fn get_suggestions(cc: &ConfigContainer, fragment: &str) -> Vec<String>;
+
+        fn get_feed_sort_strategy_values(cc: &ConfigContainer) -> Vec<i32>;
+        fn get_article_sort_strategy_values(cc: &ConfigContainer) -> Vec<i32>;
+    }
+}
+
+fn create() -> Box<ConfigContainer> {
+    Box::new(ConfigContainer(configcontainer::ConfigContainer::new()))
+}
+
+fn handle_action(
+    cc: &ConfigContainer,
+    action: &str,
+    params: Vec<String>,
+) -> bridged::CfgHandlerResult {
+    match cc.0.handle_action(action, &params) {
+        configcontainer::ConfigHandlerStatus::Success => bridged::CfgHandlerResult {
+            status: 0,
+            msg: String::new(),
+        },
+        configcontainer::ConfigHandlerStatus::InvalidCommand => bridged::CfgHandlerResult {
+            status: 1,
+            msg: String::new(),
+        },
+        configcontainer::ConfigHandlerStatus::TooFewParams => bridged::CfgHandlerResult {
+            status: 2,
+            msg: String::new(),
+        },
+        configcontainer::ConfigHandlerStatus::TooManyParams => bridged::CfgHandlerResult {
+            status: 3,
+            msg: String::new(),
+        },
+        configcontainer::ConfigHandlerStatus::InvalidParams(m) => {
+            bridged::CfgHandlerResult { status: 4, msg: m }
+        }
+    }
+}
+
+fn get_configvalue(cc: &ConfigContainer, key: &str) -> String {
+    cc.0.get_configvalue(key)
+}
+
+fn get_configvalue_as_int(cc: &ConfigContainer, key: &str) -> i32 {
+    let val = cc.0.get_configvalue(key);
+    val.parse::<i32>().unwrap_or(0)
+}
+
+fn get_configvalue_as_bool(cc: &ConfigContainer, key: &str) -> bool {
+    let val = cc.0.get_configvalue(key);
+    val == "true" || val == "yes"
+}
+
+fn get_configvalue_as_filepath(
+    cc: &ConfigContainer,
+    key: &str,
+    mut path: Pin<&mut PathBuf>,
+) -> bool {
+    let val = cc.0.get_configvalue(key);
+    if !val.is_empty() {
+        path.0 = std::path::PathBuf::from(val);
+        return true;
+    }
+    false
+}
```

Comment:
Why are conversions implemented here, rather than in `ConfigContainer` itself? Once we start adding Rust code that uses ConfigContainer, we'd need those methods in Rust too, and it's be very weird to call them via FFI module.